Google.PowerShell/BigQuery/BqTabledata.cs

```diff
@@ -0,0 +1,135 @@
+    /// schema. Also called STRUCT). Possible values for Mode include NULLABLE, REQUIRED, and 
+    /// REPEATED. Case is ignored for both Type and Mode. This command returns the new or modified 
+    /// TableSchema object.
+    /// </para>
+    /// <example>
+    ///   <code>
+    /// PS C:\> $table = Get-BqDataset “book_data” | Get-BqTable "21st_century"
+    /// PS C:\> $schema = New-BqSchema -Name "Title" -Type "STRING" -Description "Book Title"
+    /// PS C:\> $schema = $schema | New-BqSchema -Name "Author" -Type "STRING" -Description "Book Author"
+    /// PS C:\> $table.Schema = $schema
+    /// PS C:\> $table | Set-BqTable
+    ///   </code>
+    ///   <para>This will create a new schema, assign it to a table, and then send the 
+    ///   revised table to the server to be saved.</para>
+    /// </example> 
+    /// <para type="link" uri="(https://cloud.google.com/bigquery/docs/reference/rest/v2/tables)">
+    /// [BigQuery Tables]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, "BqSchema")]
+    public class NewBqSchema : PSCmdlet
+    {
+        /// <summary>
+        /// <para type="description">
+        /// The existing TableSchema that you wish to add a column to. 
+        /// If this value is not present, a new schema will be created.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipeline = true)]
+        public TableSchema InputObject { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The name of the column to be added. The name must be unique among columns in each schema.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true)]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The type of the column to be added. Possible values include STRING, BYTES, INTEGER (INT64), 
+        /// FLOAT (FLOAT64), BOOLEAN (BOOL), TIMESTAMP, DATE, TIME, DATETIME, and RECORD (STRUCT).
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true)]
+        [ValidateSet("STRING", "BYTES", "INTEGER", "INT64", "FLOAT", "FLOAT64", "BOOLEAN", "BOOL", 
+            "TIMESTAMP", "DATE", "TIME", "DATETIME", "RECORD", "STRUCT")]
+        public string Type { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// An optional description for this column.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public string Description { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The mode of the column to be added. Possible values include NULLABLE, REQUIRED, and REPEATED.
+        /// The default value is NULLABLE.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ValidateSet("NULLABLE", "REQUIRED", "REPEATED", IgnoreCase = true)]
```

Comment:
If you are going to use these values elsewhere (like in a related cmdlet), I recommend using an enum. Same for Type parameter.